app/src/main/java/com/kickstarter/viewmodels/ProjectCardHolderViewModel.java

```diff
@@ -0,0 +1,279 @@
+        .map(friends -> friends.get(0).avatar().small());
+
+      this.friendBackingViewIsHidden = this.project
+        .map(Project::isFriendBacking)
+        .map(BooleanUtils::negate);
+
+      this.friendsForNamepile = this.project
+        .filter(Project::isFriendBacking)
+        .map(Project::friends);
+
+      this.fundingUnsuccessfulTextViewIsGone = this.project
+        .map(p ->
+          !p.state().equals(Project.STATE_CANCELED) &&
+            !p.state().equals(Project.STATE_FAILED) &&
+            !p.state().equals(Project.STATE_SUSPENDED));
+
+      this.imageIsInvisible = this.project
+        .map(Project::photo)
+        .map(ObjectUtils::isNull);
+
+      this.metadataViewGroupIsGone = this.project
+        .map(p -> !p.isBacking() && !p.isStarred() && !p.isPotdToday() && !p.isFeaturedToday());
+
+      this.nameText = this.project
+        .map(Project::name);
+
+      this.notifyDelegateOfProjectClick = this.project
+        .compose(Transformers.takeWhen(this.projectClicked));
+
+      this.percentageFunded = this.project
+        .map(Project::percentageFunded)
+        .map(ProgressBarUtils::progress);
+
+      this.percentageFundedProgressBarIsGone = projectInEndedStates;
+
+      this.percentageFundedText = this.project
+        .map(Project::percentageFunded)
+        .map(NumberUtils::flooredPercentage);
+
+      this.photoUrl = this.project
+        .map(Project::photo)
+        .filter(ObjectUtils::isNotNull)
+        .map(Photo::med);
+
+      this.potdViewGroupIsGone = this.project
+        .map(p -> p.isBacking() || p.isStarred() || !p.isPotdToday() || p.isFeaturedToday());
+
+      this.projectCanceledAt = this.project
+        .filter(p -> p.state().equals(Project.STATE_CANCELED))
+        .map(Project::stateChangedAt)
+        .compose(coalesce(new DateTime()));
+
+      this.projectFailedAt = this.project
+        .filter(p -> p.state().equals(Project.STATE_FAILED))
+        .map(Project::stateChangedAt)
+        .compose(coalesce(new DateTime()));
+
+      this.projectStateViewGroupIsGone = projectInEndedStates
+        .map(BooleanUtils::negate);
+
+      this.projectSuccessfulAt = this.project
+        .filter(p -> p.state().equals(Project.STATE_SUCCESSFUL))
+        .map(Project::stateChangedAt)
+        .compose(coalesce(new DateTime()));
+
+      this.projectSuspendedAt = this.project
+        .filter(p -> p.state().equals(Project.STATE_SUSPENDED))
+        .map(Project::stateChangedAt)
+        .compose(coalesce(new DateTime()));
+
+      this.rootCategoryName = this.project
+        .map(Project::category)
+        .filter(ObjectUtils::isNotNull)
+        .map(Category::root)
+        .filter(ObjectUtils::isNotNull)
+        .map(Category::name);
+
+      this.setDefaultTopPadding = this.metadataViewGroupIsGone
+        .filter(b -> b)
+        .compose(Transformers.ignoreValues());
+
+      this.setMetadataTopPadding = this.metadataViewGroupIsGone
+        .filter(b -> !b)
```

Comment:
instead of filter should we `map` and then in the view use the boolean in the view to determine if we add the padding or remove the padding?

same for `setDefaultTopPadding`